plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/frontend.ts

```diff
@@ -204,7 +154,34 @@ const addToCartWithOptionsStore = store<
 					return true;
 				}
 
-				const { productType, quantity } = context;
+				const { productType, quantity, productId } = context;
+				const cartItems = wooState.cart?.items ?? [];
+
+				if ( productType === 'simple' ) {
+					const productObject = getProductData(
+						productId,
+						productType,
+						[],
+						[]
+					);
```

Comment:
I think it would be better to update the `getProductData()` signature so the third and fourth parameters are optional, this way we don't have to pass empty arrays, here.